### DISCUSSION TITLE
Customising cache keys for serverless lambda responses

### DISCUSSION THREAD

**Discussion by sigginjals** [vercel/vercel#7765] [Help]
Looking at this documentation: https://vercel.com/docs/concepts/edge-network/caching#serverless-functions---lambdas

It is unclear to me how the cache key is generated so I wonder if there is better documentation on this somewhere? 

If not would it be possible to set up your own custom cache key?

I'm asking because I'm experiencing false-positive cache hits, the reason for it is a bit unclear since I don't understand how the keys are generated.

**Comment by leerob** [comment]
> It is unclear to me how the cache key is generated so I wonder if there is better documentation on this somewhere?
> I'm asking because I'm experiencing false-positive cache hits, the reason for it is a bit unclear since I don't understand how the keys are generated.

From that page:

> Every deployment has a unique key used for caching based on the deployment url created at build time. This means that users will never see content from a previous deployment and there's normally no need to invalidate it.

I'd also make sure to read the section about [cacheable responses](https://vercel.com/docs/concepts/edge-network/caching#cacheable-responses).

> If not would it be possible to set up your own custom cache key?

Not currently, but this is something folks have requested and we're aware of 😄

**Reply by sigginjals** [reply]
@leerob It is still unclear what the cache key consists of.
I'm not referring to static content but rather serverless lambda responses, e.g. API routes. 
For example I have a URL that a request that looks something ike this:
`api/graphql?query=crazyTown&variables={somethingAwesome: true}`
`api/graphql?query=crazyTown&variables={somethingAwesome: false}`

These seem to be generating the same cache key, so the second request returns a false positive cache hit which results in an incorrect response.

Would it be possible to list what the cache key includes exactly? E.g. host, URL, query params, deploymentId, so that I can somehow debug why this is happening.

**Reply by sigginjals** [reply]
@leerob Any more thoughts on this?
